apps/asap-server/src/routes/research-outputs.route.ts

```diff
@@ -52,35 +53,19 @@ export const researchOutputRouteFactory = (
   researchOutputRoutes.post('/research-outputs', async (req, res) => {
     const { body } = req;
 
-    const {
-      type,
-      link,
-      title,
-      asapFunded,
-      sharingStatus,
-      usedInPublication,
-      addedDate,
-      teamId,
-    } = framework.validate('body', body, createSchema);
+    const createRequest = framework.validate('body', body, createSchema);
 
-    const researchOutput = await researchOutputController.create({
-      type,
-      link,
-      title,
-      asapFunded,
-      sharingStatus,
-      usedInPublication,
-      addedDate,
-      teamId,
-    });
+    const researchOutput = await researchOutputController.create(createRequest);
 
     res.status(201).json(researchOutput);
   });
 
   return researchOutputRoutes;
 };
-const createSchema = Joi.object({
+const createSchema = Joi.object<ResearchOutputPostRequest>({
```

Comment:
@ana-yldio @markcsims @peterstarling Thought this was neat as it ensures we keep Joi up to date with the typed request

Reply:
ResearchOutputPostRequest is the contract between the BE and FE and this function validates that its compliant. I don't understand why you would retype this? The API is restful what other resources would you create using this route? Can you provide some examples scenarios?

Reply:
Long story short. Joi doesn't allow the use of generics on the validation function so the result is untyped and lame. Raised https://asaphub.atlassian.net/browse/CRN-714 to move away from Joi as it seems like a fairly important feature of validation library.